src/core/src/InfiniteDimensionalMCMCSampler.C

```diff
@@ -27,7 +27,6 @@
 
 #include <cmath>
 #include <algorithm>
```

Comment:
Re: @roystgnr here's an example, but they're all type-conversion errors: 
<pre>
In file included from ../inc/queso/Defines.h:29:0,
                 from ../inc/queso/Environment.h:28,
                 from ../inc/queso/Miscellaneous.h:28,
                 from ../../../pbauman/src/core/src/EnvironmentOptions.C:28:
../../../pbauman/src/core/src/EnvironmentOptions.C: In constructor ‘QUESO::EnvironmentOptions::EnvironmentOptions(const QUESO::BaseEnvironment&, const char*)’:
../inc/queso/asserts.h:76:196: error: invalid static_cast from type ‘std::__cxx11::string {aka std::__cxx11::basic_string<char>}’ to type ‘type2 {aka const char (&)[1]}’
 #define queso_require_not_equal_to_msg(expr1,expr2,msg)  do { typedef decltype(expr1) type1; typedef decltype(expr2) type2; if (!((expr1 != static_cast<type1>(expr2)) && (static_cast<type2>(expr1) != expr2))) { std::cerr << "Assertion `" #expr1 " != " #expr2 "' failed.\n" #expr1 " = " << (expr1) << "\n" #expr2 " = " << (expr2) << '\n' << msg << std::endl; queso_error(); } } while(0)
                                                                                                                                                                                                    ^
../../../pbauman/src/core/src/EnvironmentOptions.C:255:3: note: in expansion of macro ‘queso_require_not_equal_to_msg’
   queso_require_not_equal_to_msg(m_env.optionsInputFileName(), "", "this constructor is incompatible with the abscense of an options input file");
</pre>

A lot of them could be fixed with making things like `""` or `"="` explicitly `std::string("")` etc (and I actually started sed'ing for those), but then there were others and I lost patience:

<pre>
In file included from ../inc/queso/Defines.h:29:0,
                 from ../inc/queso/DistArray.h:28,
                 from ../inc/queso/VectorSpace.h:28,
                 from ../inc/queso/VectorSequence.h:30,
                 from ../../../pbauman/src/basic/src/VectorSequence.C:26:
../../../pbauman/src/basic/src/VectorSequence.C: In instantiation of ‘double QUESO::BaseVectorSequence<V, M>::unifiedPositionsOfMaximum(const QUESO::ScalarSequence<double>&, QUESO::BaseVectorSequence<V, M>&) [with V = QUESO::GslVector; M = QUESO::GslMatrix]’:
../../../pbauman/src/basic/src/VectorSequence.C:2910:23:   required from here
../inc/queso/asserts.h:75:134: error: invalid static_cast from type ‘int’ to type ‘type1 {aka int&}’
 #define queso_require_equal_to_msg(expr1,expr2,msg)  do { typedef decltype(expr1) type1; typedef decltype(expr2) type2; if (!((expr1 == static_cast<type1>(expr2)) && static_cast<type2>(expr1) == expr2)) { std::cerr << "Assertion `" #expr1 " == " #expr2 "' failed.\n" #expr1 " = " << (expr1) << "\n" #expr2 " = " << (expr2) << '\n' << msg << std::endl; queso_error(); } } while(0)
                                                                                                                                      ^
../../../pbauman/src/basic/src/VectorSequence.C:495:5: note: in expansion of macro ‘queso_require_equal_to_msg’
     queso_require_equal_to_msg(recvcntsPos[0], (int) subNumPos, "failed MPI.Gather() result at proc 0 (recvcntsPos[0])");
</pre>